R/bundle.R

```diff
@@ -661,19 +683,32 @@ inferAppPrimaryDoc <- function(appPrimaryDoc, appFiles, appMode, isQuarto = FALS
 }
 
 ## check for extra dependencies congruent to application mode
-inferDependencies <- function(appMode, hasParameters, python, hasPythonRmd) {
+inferDependencies <- function(appMode, hasParameters, python, hasPythonRmd, quarto) {
   deps <- c()
-  if (grepl("\\brmd\\b", appMode)) {
+  if (appMode == "rmd-static") {
     if (hasParameters) {
       # An Rmd with parameters needs shiny to run the customization app.
       deps <- c(deps, "shiny")
     }
     deps <- c(deps, "rmarkdown")
   }
-  if (grepl("\\bshiny\\b", appMode)) {
+  if (appMode == "quarto-static") {
```

Comment:
Nit: we usually use `identical` to compare strings since `==` throws runtime errors if the LHS isn't a length 1 character vector.

```r
> foo <- NULL
> if (foo == "bar") { print("baz") }
Error in if (foo == "bar") { : argument is of length zero
```

Reply:
Chatted offline; the `appMode` is always computed by `rsconnect` and known to be a single string. Leaving as-is for now.